core/src/engine/bedrock/bedrock.ts

```diff
@@ -124,6 +138,38 @@ export function getMessages(content: Content): Message[] {
   return [{ role: "system", content: system }, ...augment, ...parts];
 }
 
+export function getMessagesFolder(
+  content: Content,
+  context: Record<string, Content>
+): Message[] {
+  const system = (content.context.system ?? [])
+    .map((s) => s.content)
+    .join("\n");
+  const history: Content[] = [];
+
+  const augment = (content.context.folder ?? [])
+    .map((c) => context[c])
+    .map<Message[]>((c) => [
+      { role: "user", content: `The contents of the file ${c.name}` },
+      { role: "assistant", content: c.prompt ?? c.response ?? "" },
+    ])
+    .flat();
+
+  const parts = history
```

Comment:
Looks like this could be extracted as a function

Reply:
Where else does it get used? (it's _subtly_ different from OpenAI's, they aren't actually interchangeable)

Reply:
https://github.com/DavidSouther/ailly/pull/43/files#diff-707f540f9750323f234f669f3370dddf51e4918c120f5002eaf90cf2d8ae0635R126-R137

Reply:
I'm going with "three strikes refactor" on this - I see where you're going, but for this PR, I want to focus on changing from `Content[]` to `(string[], Record<string, Content>)`.